File: LineManDecode.py
# by Thanatthuch Kamseng

class B64Decoder:
    CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"

    def B8(self, text, length):
        Bit8Array = []

        for i in range(0, len(text), length):
            Bit8Array.append(text hi[i:i+length])
        return Bit8Array

    def decode(self, text):
        # if in text is '=' for replace by A
       
        text = text.replace("=", "A")

        # Declare Empty string to prepare character6bit insert
        boxstring = ""
        for char in text:
            boxstring += "{:0>6b}".format(self.CHARACTERS.index(char))
        
        # Transformation B64 to B8
        B8bytes = self.B8(boxstring, 8)
        
        # Transformation B8 to Byte text and insert it. 
        bytesText = b""
        for B8 in B8bytes:
            #Decode Ascii to Base64 index and Decode it
            bytesText += bytes([int(B8, 2)])

        # the result of bytesText is:  b'iangnoW:NAM:ENIL:ta:su:nioJ'
        # Reverse text form byte-type to 'utf-8' and replace ":" by a space 
        textResult = bytesText[::-1].decode('utf-8').replace(":", " ")

        return textResult



if __name__ == "__main__":
    print("================================= Start ====================================")

    Secret = "aWFuZ25vVzpOQU06RU5JTDp0YTpzdTpuaW9K"

    SecretDecoded = B64Decoder().decode(Secret)
    print(f"\nThe result is : {SecretDecoded}")
    
    print("\nThank for consider\nMy full name: Thanatthuch Kamseng\nRole: Machine Learning Engineer\n*ContractMe*\nLine: thanatthuch38\nCall: 085-115-4196\nEmail: thanatthuchkamseng@gmail.com")
    print("================================= ENDL ====================================")

File: LineManDecode.cpp
#include <iostream>
#include <bitset>
#include <vector> 


using namespace std;


class Decoder {
    private:
        int binaryToDecimal(string n)
        {
            string num = n;
            int dec_value = 0;
            int base = 1;
        
            int len = num.length();
            for (int i = len - 1; i >= 0; i--) {
                if (num[i] == '1')
                    dec_value += base;
                base = base * 2;
            }
            return dec_value;
        }

        string set6bit(string Secret) {
            string s1, s2;
            string Override;
            static char b64table[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
                                     "abcdefghijklmnopqrstuvwxyz"
                                     "0123456789+/";
            for(int i=0; i<Secret.size(); i++){
                s1 = Secret[i];
                for(int j=0; j<sizeof(b64table); j++){
                    s2 = b64table[j];
                    if(s1 == s2){
                        bitset<6> c(j);
                        Override += c.to_string();
                    }
                };
            }
            return Override;
        }

        string set8bit(string Bits) {
            int step = 8;
            string text = "";
            string y;
            for(int i=0; i<Bits.size(); i++){
                if((i%step) == 0) {
                    string ch;
                    for(int j=i; j<i+step; j++){
                        ch +=Bits[j];
                
                    }
                    char x = static_cast<char>(binaryToDecimal(ch));
                    y = x;
                    text += y;
                    // cout << typeid(y).name() << endl;
                }
            }

            return text;
        }
    public:
        string decode(string Secret) {
            string Empty;
            string Override = set8bit(set6bit(Secret));
            int Length = Override.size();
            for(int i=0; Length>i; Length--){
                if (Override[Length] ==58) {
                    Empty += " ";
                } else {
                    Empty += Override[Length];
                }

            }
            return Empty;
        }
};



int main(){

    static string Secret = "aWFuZ25vVzpOQU06RU5JTDp0YTpzdTpuaW9K";

    Decoder step1;

    string MyResutl = step1.decode(Secret);


    cout << "======================================= By Thanatthuch Kamseng =========================================" << endl;
    cout <<"Result: " << MyResutl << endl;

    cout <<"Contrack: \nLine: Thanatthuch38\nTel: 085-115-4196\nEmail: thanatthuchkamseng@gmail.com\nEmail2: thanatthuch.cu@gmail.com"<< endl;
    cout << "================================================ Thank =================================================" << endl;

}#include <iostream>
#include <bitset>
#include <vector> 


using namespace std;


class Decoder {
    private:
        int binaryToDecimal(string n)
        {
            string num = n;
            int dec_value = 0;
            int base = 1;
        
            int len = num.length();
            for (int i = len - 1; i >= 0; i--) {
                if (num[i] == '1')
                    dec_value += base;
                base = base * 2;
            }
            return dec_value;
        }

        string set6bit(string Secret) {
            string s1, s2;
            string Override;
            static char b64table[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
                                     "abcdefghijklmnopqrstuvwxyz"
                                     "0123456789+/";
            for(int i=0; i<Secret.size(); i++){
                s1 = Secret[i];
                for(int j=0; j<sizeof(b64table); j++){
                    s2 = b64table[j];
                    if(s1 == s2){
                        bitset<6> c(j);
                        Override += c.to_string();
                    }
                };
            }
            return Override;
        }

        string set8bit(string Bits) {
            int step = 8;
            string text = "";
            string y;
            for(int i=0; i<Bits.size(); i++){
                if((i%step) == 0) {
                    string ch;
                    for(int j=i; j<i+step; j++){
                        ch +=Bits[j];
                
                    }
                    char x = static_cast<char>(binaryToDecimal(ch));
                    y = x;
                    text += y;
                    // cout << typeid(y).name() << endl;
                }
            }

            return text;
        }
    public:
        string decode(string Secret) {
            string Empty;
            string Override = set8bit(set6bit(Secret));
            int Length = Override.size();
            for(int i=0; Length>i; Length--){
                if (Override[Length] ==58) {
                    Empty += " ";
                } else {
                    Empty += Override[Length];
                }

            }
            return Empty;
        }
};



int main(){

    static string Secret = "aWFuZ25vVzpOQU06RU5JTDp0YTpzdTpuaW9K";

    Decoder step1;

    string MyResutl = step1.decode(Secret);


    cout << "======================================= By Thanatthuch Kamseng =========================================" << endl;
    cout <<"Result: " << MyResutl << endl;

    cout <<"Contrack: \nLine: Thanatthuch38\nTel: 085-115-4196\nEmail: thanatthuchkamseng@gmail.com\nEmail2: thanatthuch.cu@gmail.com"<< endl;
    cout << "================================================ Thank =================================================" << endl;
    return 0;
}
